docs/config.md

```diff
@@ -122,18 +124,304 @@ model_names: <comma-delimited-string|list>
+The unit for this field is megabytes.
+```yaml
+gpu_used_memory:
+    max: 200
+```
+
+The config above will filter out the results that their `gpu_used_memory` is larger than
+200 MBs.
+
+You can also use these keys together:
+```yaml
+gpu_used_memory:
+    max: 200
+perf_latency:
+    max: 100
+```
+
+This will filter out the results that their `gpu_used_memory` is larger than 200 MBs and
+their latency is larger than 100 milliseconds.
+
+The values described above can be specified both globally and on a per model basis.
+The global example looks like below:
+
+```yaml
+model_repository: /path/to/model-repository
+model_names:
+  - model_1
+  - model_2
+
+constraints:
+    gpu_used_memory:
+        max: 200
+```
+
+In the global mode, the constraint specified will be enforced on every model.
+If you have different constraints for each model, you can use the per model version:
+
+```yaml
+model_repository: /path/to/model-repository
+model_names:
+  model_1:
+    constraints:
+        gpu_used_memory:
+            max: 200
+  model_2:
+    constraints:
+        latency:
+            max: 50
+```
+
+## `<objective>`
+
+Objectives allow you to specify the sorting criteria for the final results.
+The fields below are supported under this object type:
+
+1. `perf_throughput`
+2. `perf_latency`
+3. `gpu_used_memory`
+4. `gpu_free_memory`
+5. `gpu_utilization`
+6. `cpu_used_ram`
+7. `cpu_available_ram`
+
+An example `objectives` that will sort the results by throughput looks like below:
+
+```yaml
+objectives:
+- perf_throughput
+```
+
+If you want to sort the results by latency, `objectives` should look like:
+
+```yaml
+objectives:
+- perf_latency
+```
+### Weighted Objectives
+
+In addition to the mode discussed above, you provide multiple values in
+the objectives key. For example:
+
+```yaml
+objectives:
+- perf_latency
+- perf_throughput
+```
+
+The above config will multiply obtained throughput and latency for each
+measurement by 0.5 (i.e. 0.5 * latency + 0.5 * throughput) and use this as
+the "score" to sort the measurements according to. This mode can be useful
+if you have a more advanced criteria upon which you want to sort the results.
+
+An extension of the above `objectives` is explicitly specifying the weights.
+For example:
+```yaml
+objectives:
+    perf_latency: 10
+    perf_throughput: 15
+```
+
+The score for each measurement will be a weighted average using the weights specified here.
+
+Similar to `<constraint>`, `<objective>` can be specified both globally and on a
+per model basis.
+
+## `<parameter>`
+
+This type can contain `concurrency` or `batch_sizes` as the possible keys. You
+can specify both or one of them. `<parameter>` must be specified under the `model_names`
+key for each model and cannot be specified globally.
+
+An example `<parameter>` looks like below:
+
+```yaml
+parameters:
+    concurrency:
+        start: 2
+        stop: 64
+        step: 8
+    batch_sizes: 1,2,3
+```
+
```

Comment:
You should write a paragraph here explaining what the parameters above will do. For example,

"These parameters will result in testing the concurrency configurations of 2, 10, 18, 26, 34, 42, 50, 58, and 64, for each of different batch sizes of 1, 2 and 3. This will result in 27 individual test runs of the model."

Reply:
Are there default values here? What is chosen if the user specifies nothing?

Reply:
If the user does not specify anything the auto config search mode will be enabled which is explained in `config_search.md`.